### PR TITLE
Asserts replacer

### PR DESCRIPTION
**Pull request by lmctv**
@reaperhulk  could this somewhat fit your ideas on #212 solution?

### CODE REVIEW

**Comment by lmctv** [comment]
As always, I'd be grateful for any naming suggestion.

As for the API, I've thought about both the natural alternatives
```
def check_condition(condition, raised, *args):
```
or
```
def check_condition(condition, *args):
```
But I'd prefer being able to have the failed test raise a custom exception, while being able to just specify the condition in an assert-like guise.

**Comment by lmctv** [comment]
@reaperhulk  I think this PR and #216 are both ready for review.

**Comment by lmctv** [comment]
Added a test for a custom exception as requested. While at it, I've corrected some other tests, and added a pynacl AssertionError, which can replace stdlib AssertionError whithout real changes to downstream expectations.

File: src/nacl/utils.py [hunk 1]
```diff
@@ -58,3 +59,24 @@ def __str__(self):
 
 def random(size=32):
     return nacl.bindings.randombytes(size)
+
+
+def ensure(cond, *args, **kwds):
+    """
+    Return if a condition is true, otherwise raise a caller-configurable
+    :py:class:`Exception`
+    :param bool cond: the condition to be chacked
```
Comment (reaperhulk) [inline src/nacl/utils.py]: checked

File: src/nacl/utils.py [hunk 2]
```diff
@@ -58,3 +59,24 @@ def __str__(self):
...
+    :param sequence args: the arguments to be passed to the exception's
+                          consructor
```
Comment (reaperhulk) [inline src/nacl/utils.py]: constructor

Comment (lmctv) [inline src/nacl/utils.py]: Corrected both typos

**Pull request merged**